src/xtgeo/grid3d/grid.py

```diff
@@ -1527,44 +1583,50 @@ def get_dualactnum_indices(self, order="C", fracture=False):
         if not self._dualporo:
             return None
 
+        assert self._dualactnum is not None
         actnumv = self._dualactnum.values.copy(order=order)
         actnumv = np.ravel(actnumv, order=order)
 
-        if not fracture:
-            actnumvm = actnumv.copy()
-            actnumvm[(actnumv == 3) | (actnumv == 1)] = 1
-            actnumvm[(actnumv == 2) | (actnumv == 0)] = 0
-            ind = np.flatnonzero(actnumvm)
-        else:
+        if fracture:
             actnumvf = actnumv.copy()
             actnumvf[(actnumv == 3) | (actnumv == 2)] = 1
             actnumvf[(actnumv == 1) | (actnumv == 0)] = 0
-            ind = np.flatnonzero(actnumvf)
-
-        return ind
+            return np.flatnonzero(actnumvf)
+        else:
+            actnumvm = actnumv.copy()
+            actnumvm[(actnumv == 3) | (actnumv == 1)] = 1
+            actnumvm[(actnumv == 2) | (actnumv == 0)] = 0
+            return np.flatnonzero(actnumvm)
 
     @deprecation.deprecated(
         deprecated_in="2.16",
         removed_in="4.0",
         current_version=xtgeo.version,
         details="Use xtgeo.Grid().gridprops instead",
     )
-    def get_gridproperties(self):
+    def get_gridproperties(self) -> GridProperties:
         """Return the :obj:`GridProperties` instance attached to the grid.
 
         See also the :meth:`gridprops` property
         """
         return self._props
 
-    def get_prop_by_name(self, name):
+    def get_prop_by_name(self, name: str) -> Optional[GridProperty]:
         """Gets a property object by name lookup, return None if not present."""
+        assert self.props is not None
```

Comment:
I somewhat, agree but. However i feel its okey in this case due to: If `self.pros is None` then the loop would fail. But there is an `return None` if there is no matach on name, so i could return none if self.props is none. But i worry abit about any potential side affects in user-space.

Reply:
Ohh, that makes sense, thanks for clarifying. Maybe a `RuntimeError` might be more appropriate then so that it at least fails more helpfully now?